### DISCUSSION TITLE
Playback of events

### DISCUSSION THREAD

**Discussion by o15a3d4l11s2** [orchetect/swift-midi#157] [Q&A] [answered]
I have a digital piano that is connected to an iPad. I want to create an application that reads the inputs from the digital piano and outputs sound on the iPad. I could not find any examples of this. Maybe there are, but I do not recognise them?

I use `manager.addInputConnection` to receive the events, but how should I play them on the iPad?

**Comment by orchetect** [comment, marked as answer]
MIDIKit only provides MIDI connections and MIDI events and abstractions. It does not produce any sound and does not contain a sampler.

I would recommend [AudioKit](https://github.com/AudioKit/AudioKit), which is a Swift library for audio playback and synthesis. I recently updated the MIDI inter-op for it to use MIDIKit on the AudioKit [v6 branch](https://github.com/AudioKit/AudioKit/tree/v6) (which is under active development).
